packages/analytics-js/src/app/RudderAnalytics.ts

```diff
@@ -133,6 +141,74 @@ class RudderAnalytics implements IRudderAnalytics<IAnalytics> {
     this.getAnalyticsInstance(writeKey).load(writeKey, dataPlaneUrl, loadOptions);
   }
 
+  /**
+   * A function to track page lifecycle events like page loaded and page unloaded
+   * @param preloadedEventsArray
+   * @param loadOptions
+   * @returns
+   */
+  // eslint-disable-next-line class-methods-use-this
+  trackPageLifecycleEvents(
+    preloadedEventsArray: PreloadedEventCall[],
+    loadOptions?: Partial<LoadOptions>,
+  ) {
+    const { trackPageLifecycle, useBeacon } = loadOptions ?? {};
+    const { events = [], enabled = false, options = {} } = trackPageLifecycle ?? {};
+    if (enabled) {
+      const visitId = generateUUID();
+      const pageLoadedTimestamp = Date.now();
+      if (events.length === 0 || events.includes(PageLifecycleEvents.PAGELOADED)) {
+        preloadedEventsArray.unshift([
+          'track',
+          PageLifecycleEvents.PAGELOADED,
+          { visitId },
+          {
+            originalTimestamp: new Date(pageLoadedTimestamp).toISOString(),
+            ...options,
```

Comment:
We should not let the user override the default `originalTimestamp` parameter. It'll mess up other things for us.
So, let's promote `...options` ahead of `originalTimestamp`.

Reply:
Addressed